cmake/shields.cmake

```diff
@@ -1,18 +1,138 @@
-if(CMAKE_SCRIPT_MODE_FILE AND NOT CMAKE_PARENT_LIST_FILE)
-  # This file was invoked as a script directly with -P:
-  # cmake -P shields.cmake
-  #
-  # Unlike boards.cmake, this takes no OUTPUT_FILE option, but
-  # SHIELD_LIST_SPACE_SEPARATED is required.
-  list(SORT SHIELD_LIST)
-  foreach(shield ${SHIELD_LIST})
-    message("${shield}")
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# Validate shields and setup shields target.
+#
+# This module will validate the SHIELD argument.
+#
+# If a shield implementation is not found for one of the specified shields an
+# error will be raised and list of valid shields will be printed.
+#
+# Outcome:
+# The following variables will be defined when this module completes:
+# - shield_conf_files: List of shield specific Kconfig fragments
+# - shield_dts_files : List of shield specific devicetree files
+# - shield_dts_fixups: List of shield specific devicetree fixups
+#
+# The following targets will be defined when this module completes:
+# - shields: when invoked a list of valid shields will be printed
+#
+# Variable dependencies:
+# - BOARD_ROOT: CMake list of board roots containing board implementations
+#
+# Module dependencies:
+# - extension module.
+
+# Check that SHIELD has not changed.
+zephyr_check_cache(SHIELD WATCH)
+
+if(SHIELD)
+  set(BOARD_MESSAGE "${BOARD_MESSAGE}, Shield(s): ${SHIELD}")
```

Comment:
Thanks for catching.
I have now split the printing in two, so that each CMake module prints its own information, like this:
```
-- Board: nrf52840dk_nrf52840
-- Shield(s): x_nucleo_idb05a1
```